src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -39,11 +40,20 @@ public class OtlpExporterOptions
         internal const string TimeoutEnvVarName = "OTEL_EXPORTER_OTLP_TIMEOUT";
         internal const string ProtocolEnvVarName = "OTEL_EXPORTER_OTLP_PROTOCOL";
 
+        internal static readonly KeyValuePair<string, string>[] StandardHeaders = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("User-Agent", UserAgentProductVersion != null ? $"{UserAgentProduct}/{UserAgentProductVersion}" : UserAgentProduct),
+        };
+
         internal readonly Func<HttpClient> DefaultHttpClientFactory;
 
         private const string DefaultGrpcEndpoint = "http://localhost:4317";
         private const string DefaultHttpEndpoint = "http://localhost:4318";
         private const OtlpExportProtocol DefaultOtlpExportProtocol = OtlpExportProtocol.Grpc;
+        private const string UserAgentProduct = "OTel-OTLP-Exporter-Dotnet";
+
+        private static readonly AssemblyName AssemblyName = typeof(OtlpExporterOptions).Assembly.GetName();
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Resources/ResourceBuilderExtensions.cs#L135 Protect inside try catch?

Reply:
Ha, got me! I tried slipping this one by you. 😆 

This was on my todo list awhile back to centralize this and apply it across the repo, I'll get on this for post 1.4. In the meantime, I applied the try/catch to just this spot.

Interesting though, I hadn't realized ResourceBuilderExtensions is actually doing things a little differently than most other places.

```csharp
typeof(Resource).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version
```

vs.

```csharp
typeof(Resource).Assembly.GetName().Version
```

Any idea why? I can look into it, but was curious if you knew off hand.

Reply:
not able to recollect the history....